openmetadata-ui/src/main/resources/ui/src/components/Explore/ExploreTree/ExploreTree.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  getAggregations,
+  getSubLevelHierarchyKey,
+  updateTreeData,
+} from '../../../utils/Explore.utils';
+import searchClassBase from '../../../utils/SearchClassBase';
+import serviceUtilClassBase from '../../../utils/ServiceUtilClassBase';
+import { getEntityIcon } from '../../../utils/TableUtils';
+import {
+  ExploreTreeNode,
+  ExploreTreeProps,
+  TreeNodeData,
+} from './ExploreTree.interface';
+
+const ExploreTreeTitle = ({ node }: { node: ExploreTreeNode }) => (
+  <Typography.Text
+    className={classNames({
+      'm-l-xs': node.data?.isRoot,
+    })}
+    data-testid={`explore-tree-title-${node.title}`}>
+    {node.title}
+  </Typography.Text>
+);
+
+const ExploreTree = ({ onFieldValueSelect }: ExploreTreeProps) => {
+  const initTreeData = searchClassBase.getExploreTree();
+  const [treeData, setTreeData] = useState(initTreeData);
+
+  const onLoadData = useCallback(
+    async (treeNode: ExploreTreeNode) => {
+      if (treeNode.children) {
+        return;
+      }
+
+      const {
+        isRoot = false,
+        currentBucketKey,
+        currentBucketValue,
+        filterField = [],
+        rootIndex,
+      } = treeNode?.data as TreeNodeData;
+
+      const searchIndex = isRoot
+        ? treeNode.key
+        : treeNode?.data?.parentSearchIndex;
+
+      const { bucket: bucketToFind, queryFilter } = getSubLevelHierarchyKey(
+        rootIndex === SearchIndex.DATABASE,
+        currentBucketKey as EntityFields,
+        currentBucketValue
+      );
+
+      const res = await getAggregateFieldOptions(
+        searchIndex as SearchIndex,
+        bucketToFind,
+        '',
+        JSON.stringify(queryFilter)
+      );
+      const aggregations = getAggregations(res.data.aggregations);
+      const buckets = aggregations[bucketToFind].buckets;
+      const isServiceType = bucketToFind === EntityFields.SERVICE_TYPE;
+      const isEntityType = bucketToFind === EntityFields.ENTITY_TYPE;
+
+      const sortedBuckets = [...buckets].sort((a, b) =>
+        a.key.localeCompare(b.key, undefined, { sensitivity: 'base' })
+      );
+
+      const children = sortedBuckets.map((bucket) => {
+        let logo = <></>;
+        const title = (
+          <div className="d-flex justify-between">
+            <Typography.Text className="m-l-xs">
+              {isEntityType ? getEntityNameLabel(bucket.key) : bucket.key}
+            </Typography.Text>
+            {isEntityType && <span>{getCountBadge(bucket.doc_count)}</span>}
+          </div>
+        );
+        if (isEntityType) {
+          logo = getEntityIcon(bucket.key, 'service-icon w-4 h-4');
+        } else if (isServiceType) {
+          const serviceIcon = serviceUtilClassBase.getServiceLogo(bucket.key);
+          logo = (
+            <img
+              alt="logo"
+              src={serviceIcon}
+              style={{ width: 18, height: 18 }}
+            />
+          );
+        }
+
+        return {
+          title: title,
+          key: uniqueId(),
```

Comment:
this will every time give you a new key, and it will re-render the full component even if 1 value is changed!

Reply:
No, it will not. This will only create children once when new items are fetched.
It will not update the existing items.